text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,337 @@
+
+Investigation should take place to discern if this could be changed without breaking the public API and/or existing stack resources. If not, is creating a new construct and deprecating the old one at the top level a viable strategy for constructs with planned breaking changes going forward? Is it appropriate to deprecate a portion of a construct (constructor properties, etc) that would require resource replacement? Can we guide users through replacement of constructs that are being deprecated to make sure they don't experience interruptions in service? Because of these unknowns, it is unlikely that this will be included in the 2.0 release.
+
+- [Change Logical IDs to Avoid Potential Collisions](https://github.com/aws/aws-cdk/issues/6421)
+
+Similar to aws/aws-cdk#1687, changing logical ID logic introduces a lot of problems for existing users. Whether `LogicalIds 2.0` can be implemented behind a feature flag should be investigated to avoid breaking existing apps.
+
+- [AliasOptions and VersionOptions should not extend EventInvokeConfigOptions](https://github.com/aws/aws-cdk/issues/6966)
+
+More information needed. @eladb
+
+- [Remove Support for Docker Assets with Parameters]()
+
+More information needed. @eladb
+
+# Drawbacks
+
+The main drawback of this staged release strategy is the complexity we will incur within our build tooling to be more version aware. Though doing things like selectively including or excluding deprecated APIs based on vurrent version allows us to phase the release of breaking changes without having to maintain a long lived branch for the upcoming release, the potential for bugs in the functionality of these features could have downsides to users. If a deprecated API is removed from a module when it shouldn't be, and that module is published, it will break user's applications. Though this could of course be remedied in a patch release.
+
+If a deprecated API isn't removed from a version that it was inteded to be, and users start relying on the presence of that API, then by fixing the bug and removing the API in future versions we potentionally introduce breaking changes that weren't intended by us or expected by the users.
+
+Because of this reason, automated verification of our different builds and the presence/absence of APIs and feature flags is, as always, of high importance.
+
+The cost of migration for users varies depending on the state of the application being migrated. All users upgrading to v2 will have to perform the following steps:
+
+1. Update Dependencies - Changing from referencing the old `@aws-cdk/` packages to the new `aws-cdk-lib` package in the corresponding target language's package manifest (package.json, .sln, pom.xml, setup.py).
+2. Update Imports - Since the current module structure is being changed, import statements in user's code will need to change to accommodate the new structure. This will be automated wherever needed with the [import rewriter](#import-rewriter).
```

Comment:
Just a niggle, but an automatic rewriter is not guaranteed to work in the presence of removing deprecated APIs.

If you were using deprecated stuff, it's gone now. No way to rewrite around that.